rustls/src/crypto/aws_lc_rs/tls12.rs

```diff
@@ -384,3 +384,38 @@ fn gcm_iv(write_iv: &[u8], explicit: &[u8]) -> Iv {
 
     Iv::new(iv)
 }
+
+struct Tls12Prf(&'static tls_prf::Algorithm);
+
+impl Prf for Tls12Prf {
+    fn for_secret(&self, output: &mut [u8], secret: &[u8], label: &[u8], seed: &[u8]) {
+        // safety:
+        // - [1] is safe because our caller guarantees `secret` is non-empty; this is
+        //   the only documented error case.
+        // - [2] is safe in practice because the only failure from `derive()` is due
+        //   to zero `output.len()`; this is outlawed at higher levels
+        let result = tls_prf::Secret::new(self.0, secret)
```

Comment:
Nit: I don't like the name `result` for this because I feel like bindings called `result` are usually of type `Result`. Maybe `next_secret` or `new`?